src/osd/ExtentCache.h

```diff
@@ -297,6 +304,19 @@ class ExtentCache {
 		extoff, extlen);
 	    }
 	    final_extent->link(*this, pin);
+	  } else if (action.action == update_action::RESERVE_EXT) {
+	    bufferlist bl;
+            bl.substr_of(
+                *(ext->bl),
+                extoff - ext->offset,
+                extlen);
+	    final_extent = new ExtentCache::extent(
+                extoff,
+	        bl);
+	    final_extent->recommend_reserve = true;
+	    assert(action.from != boost::none);
+	    final_extent->from = *action.from;
+	    final_extent->link(*this, *ps);
```

Comment:
Popping up a level, is it worth splitting one of these extents simply to mark it reserved?

Reply:
Actually, it would probably make more sense to make this a flag in the constructor or on reserve_extents_for_rmw.  As implemented, this patch specifically reserves the extents which are not in the cache, it probably makes more sense to just blanket reserve the whole thing.

Reply:
> As implemented, this patch specifically reserves the extents which are not in the cach

If I did not misunderstand your meaning, here I want to specifically reserve the *to_read* extents that already reserved in the cache when invoke the reserve_extents_for_rmw().

The *to_read* extents must the begin stripe or the end stripe or both of them, I means it will be two stripes at most, If we reserve whole thing, I'm afraid that the memory may lost control.. make sense?

Reply:
I guess I'm not sure what the goal here is.  What workload are you trying to improve?

Reply:
Non-stripe aligned sequential writing or hot stripes that non-stripe aligned access.

Reply:
I guess I'm not sure why we wouldn't want to cache the other stripes as well assuming that the cache is underutilized given that we already have the bytes in memory.  Is the assumption that the edge stripes in particular are likely to be edge stripes in the future?

Reply:
> Is the assumption that the edge stripes in particular are likely to be edge stripes in the future?

I think it is yes for non-stripe aligned sequential writing, then hope there will be some hot stripes..

Reply:
It would surprise me if caching the middle extents as well didn't further improve performance.

Reply:
Oh sorry, there may be some mistakes, Let me explain it a bit more explicitly : 
1. If we cache the edge stripes, this will be helpful for non-stripe aligned sequential writing; 
2. if we caching the middle extents I think this also may improve performance on some certain circumstance.. 

The cached stripes is intend to avoid rmw, So what I accurately want to express is the edge stripes to be cached maybe has more (or less) value to cache than the middle :D

Reply:
Do we have any numbers on the relative performance?  To that end, adding perf counters to count hits and misses would be helpful for evaluating efficacy.